x/merkledb/trie_test.go

```diff
@@ -467,21 +467,19 @@ func Test_Trie_HashCountOnBranch(t *testing.T) {
 	dbTrie, err := getBasicDB()
 	require.NoError(err)
 	require.NotNil(dbTrie)
-	trieIntf, err := dbTrie.NewView(nil)
+	trieIntf, err := dbTrie.NewView(context.Background(), nil)
 	require.NoError(err)
 	trie := trieIntf.(*trieView)
 
 	// force a new node to generate with common prefix "key1" and have these two nodes as children
 	_, err = trie.insert(newPath([]byte("key12")), maybe.Some([]byte("value12")))
 	require.NoError(err)
-	require.NoError(trie.calculateNodeIDs(context.Background()))
-	oldCount := dbTrie.metrics.(*mockMetrics).hashCount
 	_, err = trie.insert(newPath([]byte("key134")), maybe.Some([]byte("value134")))
 	require.NoError(err)
 	// only hashes the new branch node, the new child node, and root
 	// shouldn't hash the existing node
 	require.NoError(trie.calculateNodeIDs(context.Background()))
-	require.Equal(oldCount+3, dbTrie.metrics.(*mockMetrics).hashCount)
```

Comment:
why did we change this?

Reply:
I can't easily split the calculations up into two parts since it only will calculate all values once